applications/serial_lte_modem/doc/slm_description.rst

```diff
@@ -0,0 +1,1552 @@
+         #XUDPSEND: 15
+         OK
+
+         #XUDPRECV: 1, 13
+         Hello, UDP#3!
+         #XUDPRECV: 1, 13
+         Hello, UDP#4!
+         #XUDPRECV: 1, 13
+         Hello, UDP#5!
+
+         **AT#XUDPSEND=1,"UDP3/4/5 received"**
+         #XUDPSEND: 17
+         OK
+
+   #. Observe the output of the Python script::
+
+         $ python client_udp.py
+
+         Sending: 'Hello, UDP#1!
+         Sending: 'Hello, UDP#2!
+         UDP1/2 received
+         ('000.000.000.00', 1234)
+         Sending: 'Hello, UDP#3!
+         Sending: 'Hello, UDP#4!
+         Sending: 'Hello, UDP#5!
+         UDP3/4/5 received
+         ('000.000.000.00', 1234)
+         Closing connection
+
+   #. Close the socket.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XUDPSVR=0**
+         #XUDPSVR: stopped
+         OK
+
+#. Test the UDP server with UDP proxy service in data mode (note that these commands are available only if :option:`CONFIG_SLM_UDP_PROXY` is defined):
+
+   a. Create a UDP server and read information about the current state.
+      Replace *1234* with the correct port number.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XUDPSVR=2,**\ *1234*
+         #XUDPSVR: 1 started
+         OK
+
+         **AT#XUDPSVR?**
+         #XUDPSVR: 1, 1
+         OK
+
+   #. Run the :file:`client_udp.py` script to start sending data to the server.
+
+   #. Observe that the server starts receiving data.
+      Acknowledge the received data.
+
+      .. parsed-literal::
+         :class: highlight
+
+         Hello, UDP#1!Hello, UDP#2!\ **UDP1/2 received**
+         Hello, UDP#3!Hello, UDP#4!Hello, UDP#5!\ **UDP3/4/5 received**
+
+   #. Observe the output of the Python script::
+
+         $ python client_udp.py
+
+         Sending: 'Hello, UDP#1!
+         Sending: 'Hello, UDP#2!
+         UDP1/2 received
+         ('000.000.000.00', 1234)
+         Sending: 'Hello, UDP#3!
+         Sending: 'Hello, UDP#4!
+         Sending: 'Hello, UDP#5!
+         UDP3/4/5 received
+         ('000.000.000.00', 1234)
+         Closing connection
+
+   #. Stop the server.
+
+      .. parsed-literal::
+         :class: highlight
+
+         **AT#XUDPSVR=0**
+         #XUDPSVR: stopped
+         OK
+
+TLS server
+~~~~~~~~~~
+
+The TLS server role is currently not supported.
+
+.. parsed-literal::
+   :class: highlight
+
+   **AT#XSOCKET=1,1,1,16842753**
+   #XSOCKET: (D)TLS Server not supported
+   ERROR
+
+   **AT#XTCPSVR=1,3443,16842753**
+   #XTCPSVR: TLS Server not supported
+   ERROR
+
+DTLS server
+~~~~~~~~~~~
+
+The DTLS server role is currently not supported (modem limitation).
+
+.. parsed-literal::
+   :class: highlight
+
+   **AT#XSOCKET=1,2,1,16842755**
+   #XSOCKET: (D)TLS Server not supported
+   ERROR
+
+DNS lookup
+~~~~~~~~~~
+
+1. Look up the IP address for a host name.
+
+   .. parsed-literal::
+      :class: highlight
+
+      **AT#XGETADDRINFO="www.google.com"**
```

Comment:
I get an error for that command - does it depend on some other settings, or on the network?

Reply:
It seems like something wrong in the network.
Can you try below command before connecting to network?

AT%XEPCO=0

Reply:
AT#XGETADDRINFO="www.google.com"
#XGETADDRINFO: 216.58.197.164
OK
Make sure network is attached properly.

Reply:
Uh ... ok. We're missing the step that users actually need to connect to the network.
That's just "AT+CFUN=1", or is anything else required?

Reply:
Most SLM services requires network attached, such as TCP/IP, FTP, MQTT, HTTP. You can check the network state by AT+CFUN? , or set  indicator states by AT+CIND=1,1,1

The example enables service and message indicators:

AT+CIND=1,0,1
OK

The example notification indicates that the device is in service:

+CIND: "service",1

https://infocenter.nordicsemi.com/topic/ref_at_commands/REF/at_commands/mob_termination_ctrl_status/cind_set.html